File: shopping_list.js
// Shopping List Program in JavaScript
// Created by: [Your Name]
// Description: A simple program using arrays, user input, and conditionals.

const prompt = require("prompt-sync")(); // For Node.js user input
let shoppingList = [];

console.log("Enter up to 5 items to buy:");

// Step 1: Ask the user to enter up to 5 items
for (let i = 0; i < 5; i++) {
  let item = prompt("> ").trim();
    if (item === "") break; // Stop if user presses Enter without typing
      shoppingList.push(item);
      }

      // Step 2: Print the complete shopping list
      console.log("\nYour shopping list:");
      console.log(shoppingList.join(", "));

      // Step 3: Count how many items were entered
      console.log(`\nYou entered ${shoppingList.length} item(s).`);

      // Step 4 (Extra Challenge): Search feature
      let searchItem = prompt("\nSearch for an item: ").trim();

      if (shoppingList.includes(searchItem)) {
        console.log(`${searchItem} is in your shopping list!`);
        } else {
          console.log(`${searchItem} is NOT in your shopping list.`);
          }

File: shopping_list.py
# Shopping List Program
# Created by: [Your Name]
# Date: [Date]
# Description: A simple shopping list program using lists, user input, and conditionals.

# Step 1: Ask the user to enter up to 5 items
print("Enter up to 5 items to buy:")

shopping_list = []  # Empty list to store items

for i in range(5):
    item = input(f"> ").strip()
        if item:  # Only add if user enters something
                shopping_list.append(item)
                    else:
                            break  # Stop if user presses Enter without typing anything

                            # Step 2: Print the complete shopping list
                            print("\nYour shopping list:")
                            print(", ".join(shopping_list))

                            # Step 3: Count and display how many items were entered
                            print(f"\nYou entered {len(shopping_list)} item(s).")

                            # Step 4 (Extra Challenge): Search feature
                            search = input("\nSearch for an item: ").strip()

                            if search in shopping_list:
                                print(f"{search} is in your shopping list!")
                                else:
                                    print(f"{search} is NOT in your shopping list.")
